### PR TITLE
Kotlin maven plugin to 1.8.20

### PR DESCRIPTION
**Pull request by PetrakovichVictoria**
[KT-56317](https://youtrack.jetbrains.com/issue/KT-56317) [Docs][Build Tools] kotlin-maven-plugin + kapt - allow aptMode to be set according to docs
[KT-56316](https://youtrack.jetbrains.com/issue/KT-56316) [Docs][Build Tools] Add components.xml to automatically compile kotlin maven projects

### CODE REVIEW

**Comment by PetrakovichVictoria** [comment]
Initial PR https://github.com/JetBrains/kotlin-web-site/pull/3496 is approved by a developer. This PR needs only tech writers' review.

**Comment by PetrakovichVictoria** [comment]
@koshachy, @danil-pavlov, any of you:)

**Review by sarahhaggarty — COMMENTED** [review]
Looks good! Just a few minor suggestions 😃

File: docs/topics/kapt.md [hunk 1]
```diff
@@ -302,7 +302,7 @@ Add an execution of the `kapt` goal from kotlin-maven-plugin before `compile`:
 <execution>
     <id>kapt</id>
     <goals>
-        <goal>kapt</goal>
+        <goal>kapt</goal> <!-- You can skip <goals> element if you enable extensions for the plugin -->
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/kapt.md:297]: It was intentional: I tried to minimize the length of the comment (cause this line to be not scrollable). We discussed this with Andrey some time ago, he told me we could skip articles in comments when it doesn't play the leading role. But I checked it on the staging, I'll rearrange this comment to get rid of scrolling.

File: docs/topics/maven.md [hunk 2]
```diff
@@ -79,10 +79,20 @@ The Kotlin Maven Plugin needs to be referenced to compile the sources:
 </build>
 ```
 
+Starting from Kotlin 1.8.20, you can replace the whole `<executions>` element above with `<extensions>true</extensions>`. 
+Enabling extensions automatically adds the `compile`, `test-compile`, `kapt`, and `test-kapt` executions to your build, 
+bound to their appropriate [lifecycle phases](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html). 
+If you need to configure an execution, you need to specify its id; see an example in the following section.
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/maven.md:85]: The first

File: docs/topics/maven.md [hunk 3]
```diff
@@ -92,11 +102,12 @@ making sure that the `kotlin` plugin comes before the `maven-compiler-plugin` in
             <groupId>org.jetbrains.kotlin</groupId>
             <artifactId>kotlin-maven-plugin</artifactId>
             <version>${kotlin.version}</version>
+            <extensions>true</extensions> <!-- You can set this option to automatically take information about lifecycles -->
             <executions>
                 <execution>
                     <id>compile</id>
                     <goals>
-                        <goal>compile</goal>
+                        <goal>compile</goal> <!-- You can skip <goals> element if you enable extensions for the plugin -->
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/maven.md]: Fixed

File: docs/topics/maven.md [hunk 4]
```diff
@@ -107,7 +118,7 @@ making sure that the `kotlin` plugin comes before the `maven-compiler-plugin` in
                 </execution>
                 <execution>
                     <id>test-compile</id>
-                    <goals> <goal>test-compile</goal> </goals>
+                    <goals> <goal>test-compile</goal> </goals> <!-- You can skip <goals> element if you enable extensions for the plugin -->
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/maven.md]: Fixed

File: docs/topics/maven.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/maven.md]: Fixed

**Review by sarahhaggarty — APPROVED** [review]

**Pull request merged**